cosmos/ingestion/ingest/ingest.py

```diff
@@ -196,164 +199,193 @@ def raise_timeout(var1, var2):
+        logger.info(f"{pdf_name}: {all_labels}")
+
+        # for each row in the original_tables_df that has a table_label
+        for index, row in original_tables_df.iterrows():
+            table_label = row['table_label']
+            if table_label:
+                # collect contexts from doc_df
+                contexts = []
+                for words in all_doc_words:
+                    # mark where each label occurs in list of all doc words
+                    # allow for any character beside the table label e.g. Table 2. or Table 2,
+                    try:
+                        indices = [
+                                j
+                                for j in range(len(words))
+                                if re.match(table_label+r'\S', ' '.join(words[j:j + label_length]))
+                                # if table_label == ' '.join(words[j:j + label_length])
+                        ]
+
+                        logger.debug(f'indices: {indices}')
+
+                        # iterate over list in indices column to extract words before labels in content
+                        logger.debug('getting semantic context')
+                        prefixes = [
+                                ' '.join(words[i - spans:i])
+                                if i - spans >= 0
+                                else ' '.join(words[0:i])
+                                for i in indices
+                        ]
+                        logger.debug(f'prefixes: {prefixes}')
+                        # iterate over list in indices column to extract words after labels in content
+                        suffixes = [
+                                ' '.join(words[i + label_length:i + label_length + spans])
+                                if i + label_length + spans <= len(words)
+                                else ' '.join(words[i + label_length:len(words)])
+                                for i in indices
+                        ]
+                        logger.debug(f'suffixes: {suffixes}')
+
+                        logger.debug(f'list(zip(prefixes, suffixes)):{list(zip(prefixes, suffixes))}')
+                        # join prefixes and suffixes together
+                        contexts.append(' '.join([
+                                prefix + ' ' + table_label + ' ' + suffix
+                                for prefix, suffix in list(zip(prefixes, suffixes))
+                                ]
+                            )
+                        )
+                        logger.debug(f'contexts: {contexts}')
+
+                    except Exception as e:
+                        logger.info(f"{pdf_name} failed fetching context indices for '{table_label}':\n {e}")
+                        logger.info(f'table row:\n {row}')
+
+                # add all context and content to output column
+                tables_df.at[index, context_column] = row['content'] + ' ' + ' '.join(contexts).strip()
```

Comment:
I keep going back and forth on this, but I think that it makes more sense for this column to _just_ be mentions coming from the text, rather then being `content` concatenated with the context mentions. I think it's cleanest if we keep the three fields (`content`, `content_caption`, `content_enriched`) non-overlapping within the dataframe, and then combine the fields as needed for retrievability (e.g. when we ingest into Elasticsearch).

..of course that makes the name `content_enriched` a bad one. Maybe `context_from_text`? `text_mentions`?

What do you think?

Reply:
Yes that's totally fine with me. I agree it's more efficient and it should be simple to combine when needed.